app/models/manageiq/providers/google/cloud_manager/event_catcher/stream.rb

```diff
@@ -0,0 +1,62 @@
+# Helper class responsible for polling a Google Pubsub topic and retreiving
+# event type messages. Parsing of the message is deferred to the caller of
+# #poll.
+class ManageIQ::Providers::Google::CloudManager::EventCatcher::Stream
+  def initialize(ems)
+    @ems = ems
+  end
+
+  # Start capturing events
+  def start
+    @collecting_events = true
+  end
+
+  # Stop capturing events
+  def stop
+    @collecting_events = false
+  end
+
+  # Poll for events (blocks forever until #stop is called)
+  def each_batch
+    while @collecting_events
+      yield events.map { |e| JSON.parse(e.message['data']) }
```

Comment:
I considered that, but I am not sure it's needed. Right now this bubbles up an exception which kills the worker thread (and the thread is then restarted); I can change it to catch the exception but I'm unsure how it would handle it differently (other than logging and ignoring).  Given that bad json coming in would have to be a misconfiguration or a severe service issue at Google's side, I'm not sure it's worth the extra complication.

What do you think?